simpeg/potential_fields/magnetics/simulation.py

```diff
@@ -1524,8 +1524,16 @@ def projectFields(self, u):
             \text{TMI} = \vec{B}_s \cdot \hat{B}_0
 
         """
-        # TODO: There can be some different tyes of data like |B| or B
-        components = self.survey.components
+        # Get components for all receivers, assuming they all have the same components
+        try:
+            components = self._get_components()
+        except NotImplementedError as e:
+            e.args = (
+                "Found receivers with different set of components in the survey. "
+                "Calling `projectFields` with such type of survey is not currently "
+                "supported.",
+            )
+            raise e
```

Comment:
Can we issue an error on the survey class instead? If a survey has multiple receivers with different locations, accessing the `survey.components` will issue an error (in addition to it being deprecated)?

Reply:
I'd like to keep the error messages in the simulation methods. Mainly because after we remove the `survey.components` property the simulation method would still work without any other change.

We can also make `survey.components` to raise error in case that receivers have different components. What do you think?

Reply:
I know it’s a little rough due to the current mutability of surveys, but would it better to validate the survey on assignment to this simulation? Just check if a survey exists with multiple sources or receivers and fail the validation. (I expect that this type of mutability is very rare to encounter, as once surveys are assigned to a simulation, I doubt users change the surveys.)

Reply:
Only change would be to update it on the simulation’s survey setter, not on `__init__`.